java/src/jmri/jmrix/cmri/serial/serialmon/SerialMonFrame.java

```diff
@@ -28,80 +58,101 @@ protected String title() {
+     */  
+    public void initializePacketFilters()
+    {
+        // get all configured nodes
+        SerialNode node = (SerialNode) _memo.getTrafficController().getNode(0);
+        int index = 1,
+            pktTypeIndex = 0;
+        
+        while (node != null)
+        {
+         // Set all nodes and packet types to be monitored
+         //-----------------------------------------------
+	 do
+         {
+            node.setMonitorPacketBit(pktTypeIndex, true);
+            pktTypeIndex++;
+         } while ( pktTypeIndex < SerialFilterFrame.numMonPkts);
 
-    @Override
- 
-    public synchronized void message(SerialMessage l) {  // receive a message and log it
-        // check for valid length
-        if (l.getNumDataElements() < 2) {
-            nextLine("Truncated message of length " + l.getNumDataElements() + "\n",
-                    l.toString());
-            return;
-        } else if (l.isPoll()) {
-            nextLine("Poll ua=" + l.getUA() + "\n", l.toString());
-        } else if (l.isXmt()) {
-            StringBuilder sb = new StringBuilder("Transmit ua=");
-            sb.append(l.getUA());
-            sb.append(" OB=");
-            for (int i = 2; i < l.getNumDataElements(); i++) {
-                sb.append(Integer.toHexString(l.getElement(i) & 0x000000ff));
-                sb.append(" ");
-            }
-            sb.append("\n");
-            nextLine(new String(sb), l.toString());
-        } else if (l.isInit()) {
-            StringBuilder sb = new StringBuilder("Init ua=");
-            sb.append(l.getUA());
-            sb.append(" type=");
-            sb.append((char) l.getElement(2));
-            int len = l.getNumDataElements();
-            if (len >= 5) {
-                sb.append(" DL=");
-                sb.append(l.getElement(3) * 256 + l.getElement(4));
-            }
-            if (len >= 6) {
-                sb.append(" NS=");
-                sb.append(l.getElement(5));
-                sb.append(" CT: ");
-                for (int i = 6; i < l.getNumDataElements(); i++) {
-                    sb.append(Integer.toHexString(l.getElement(i) & 0x000000ff));
-                    sb.append(" ");
-                }
-            }
-            sb.append("\n");
-            nextLine(new String(sb), l.toString());
-        } else {
-            nextLine("unrecognized cmd: \"" + l.toString() + "\"\n", "");
-        }
+         node = (SerialNode) _memo.getTrafficController().getNode(index);
+         index ++;
+         pktTypeIndex = 0;
+	}
     }
 
-/*
-  /********************
-     Transmit Packets
-  *********************
+    String newline = System.getProperty("line.separator");
+
+    /**
+    * Open the node/packet filter window
+    */
+    public void openPacketFilterPerformed(ActionEvent e) 
+    {
+	// create a SerialFilterFrame
+	SerialFilterFrame f = new SerialFilterFrame(_memo);
+	try {
+		f.initComponents();
+            }
+	catch (Exception ex) {
```

Comment:
Catch only the specific exceptions expected to be thrown here. See http://jmri.org/help/en/html/doc/Technical/RP.shtml on how to catch exceptions. See also http://jmri.org/help/en/html/doc/Technical/Logging.shtml for how to log them.

Reply:
This form is used in many other classes.